### PR TITLE
lsm_local_disk_health_status_get: Add NVMe health check

### PR DESCRIPTION
**Pull request by tasleson**
Add ability to check the health status for NVMe devices.
We are using the get log page admin command with log page
2 (smart/health information) and checking the critical warning
return byte.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

### CODE REVIEW

**Comment by tasleson** [comment]
@tbzatek Please review this when you have a chance, thanks!

**Review by tbzatek — APPROVED** [review]
Good to go I guess, minor notes added.

File: c_binding/libnvme.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    struct smart_data data;
+
+    struct nvme_admin_cmd cmd = {
+        .opcode = 0x02,     // nvme admin get log page
+        .nsid = 0xffffffff, // All namespaces
+        .addr = (uint64_t)&data,
+        .data_len = sizeof(data),
+        .cdw10 = cdw10,
+        .cdw11 = number_dword_upper,
+        // 12 - 14 == are zero
```
[the remark this answers is not in the document]
Comment (tasleson) [reply c_binding/libnvme.c:78]: I'll explicitly set these to zero to be safe.

File: c_binding/libnvme.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    int rc = ioctl(fd, NVME_IOCTL_ADMIN_CMD, &cmd);
+    if (0 == rc) {
+        /* If any bits are set we are calling this a fail */
+        *health_status = (data.critical_warning == 0)
+                             ? LSM_DISK_HEALTH_STATUS_GOOD
+                             : LSM_DISK_HEALTH_STATUS_FAIL;
+        return LSM_ERR_OK;
+    }
+    int errno_cpy = errno;
+    snprintf(err_msg, _LSM_ERR_MSG_LEN, "Unexpected return from ioctl %d (%s)",
```
[the remark this answers is not in the document]
Comment (tasleson) [reply c_binding/libnvme.c:91]: Thanks for this, I should have looked into this myself.

**Pull request merged**